### PR TITLE
Quest roles BE 

### PR DESCRIPTION
**Pull request by vidvidvid**
## Overview

**What features/fixes does this PR include?**

Added `quest_roles` needed for #1034 
Added `basic` flag to determine which roles to display during onboarding and which to display when selecting roles for quests.

(Excuse me for multiple migrations, some of which were unnecessary)

paired-with: @dan13ram

### CODE REVIEW

**Comment by vidvidvid** [comment]
@dan13ram please confirm if i included everything i should have included in regards to BE changes in this pr :)

File: packages/web/graphql/getQuests.ts [hunk 1]
```diff
@@ -62,6 +62,17 @@ gql`
   }
 `;
 
+// eslint-disable-next-line @typescript-eslint/no-unused-expressions
```
Comment (dan13ram) [inline packages/web/graphql/getQuests.ts:65]: @vidvidvid  This is not necessary in this PR but looks like it doesn't make a big difference.
But you may also wanna include changes in `packages/backend/src/handlers/actions/quests/createQuest/createQuest.ts` as well.

Comment (dan13ram) [reply packages/web/graphql/getQuests.ts:65]: also I'm not able to find changes related to `quest_completion` and `quest` table relationship that we discussed over chat. Would that be included in a future PR ?

**Review by alalonde — CHANGES_REQUESTED** [review]
can you consolidate the migrations? From what I can see there should only be 3. You can test them locally via `hasura migrate apply ...`

**Comment by vidvidvid** [comment]
rebasing this branch onto develop and running `yarn update-schema` completely messes up the schema`.graphql`. any idea why is that happening? @dysbulic @dan13ram @alalonde ?

**Comment by dysbulic** [comment]
> rebasing this branch onto develop and running `yarn update-schema` completely messes up the schema`.graphql`.

@vidvidvid, have you run `yarn && yarn docker:clean && docker-compose up --build` to rebuild the database? Your existing db might have migrations in it that are out of sync given the large number of migrations in #943.

When you say it messes up `schema.graphql`, what do you mean? What sorts of errors are you getting?

**Comment by vidvidvid** [comment]
> @vidvidvid, have you run `yarn && yarn docker:clean && docker-compose up --build` to rebuild the database? Your existing db might have migrations in it that are out of sync given the large number of migrations in #943.

Not yet, I'll do that and let you know how it works.

> When you say it messes up `schema.graphql`, what do you mean? What sorts of errors are you getting?

It builds a completely different structure, as if it didn't take into the consideration all of your migrations.

**Comment by vidvidvid** [comment]
@dysbulic `yarn && yarn docker:clean && docker-compose up --build` fails at `yarn`:

```
vidtopolovec@Vids-MBP TheGame % yarn
yarn install v1.19.0
[1/5] 🔍  Validating package.json...
[2/5] 🔍  Resolving packages...
warning Resolution field "typescript@4.5.4" is incompatible with requested version "typescript@^3.7.3"
warning Resolution field "typescript@4.5.4" is incompatible with requested version "typescript@^2.6.2"
warning Resolution field "typescript@4.5.4" is incompatible with requested version "typescript@^2.6.2"
success Already up-to-date.
$ lerna run prepare && husky install
lerna notice cli v4.0.0
lerna info versioning independent
lerna info Executing command in 1 package: "yarn run prepare"
lerna ERR! yarn run prepare exited 1 in '@metafam/web'
lerna ERR! yarn run prepare stdout:
yarn run v1.19.0
$ yarn generate
$ graphql-codegen --config=codegen.yml
[11:25:29] Parse configuration [started]
[11:25:30] Parse configuration [completed]
[11:25:30] Generate outputs [started]
[11:25:30] Generate ./graphql/autogen/types.ts [started]
[11:25:30] Load GraphQL schemas [started]
[11:25:30] Load GraphQL schemas [failed]
[11:25:30] → Failed to load schema
[11:25:30] Generate ./graphql/autogen/types.ts [failed]
[11:25:30] → Failed to load schema
[11:25:30] Generate outputs [failed]
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
lerna ERR! yarn run prepare stderr:
Something went wrong
error Command failed with exit code 1.
error Command failed with exit code 1.
lerna ERR! yarn run prepare exited 1 in '@metafam/web'
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
```

**Comment by dysbulic** [comment]
> Does `yarn generate` fail because the `schema.graphql` is not valid?

It is possible… Try running `yarn generate`. I added a `DEBUG=1` environment variable to that command that should cause it to print the file and line where it is failing.

**Comment by vidvidvid** [comment]
@dysbulic I had a missing query in the schema, `yarn generate` worked after i added it 👍

**Comment by vidvidvid** [comment]
> can you consolidate the migrations? From what I can see there should only be 3. You can test them locally via `hasura migrate apply ...`

Will look into it 👍

**Comment by vidvidvid** [comment]
@alalonde I consolidated the migrations, but I'm not sure how to merge [this](https://github.com/MetaFam/TheGame/blob/ff49200507303c3ee113ad01e894a1501be4848e/hasura/migrations/1641640073308_create_table_public_quest_role/up.sql) and [that](https://github.com/MetaFam/TheGame/blob/ff49200507303c3ee113ad01e894a1501be4848e/hasura/migrations/1642342313101_alter_table_public_quest_role_drop_constraint_quest_role_quest_id_key/up.sql), could you help me out with that or could we perhaps just push this?

File: hasura/metadata/actions.graphql [hunk 2]
```diff
@@ -73,6 +73,7 @@ input CreateQuestInput {
   externalLink : String
   repetition : QuestRepetition_ActionEnum
   cooldown : Int
+  roles_id : [uuid]!
```
Comment (alalonde) [inline hasura/metadata/actions.graphql:76]: let's make this camelcase for consistency

Comment (alalonde) [reply hasura/metadata/actions.graphql:76]: `roleIds` ?

**Review by alalonde — COMMENTED** [review]
You can have multiple lines in a given migration, both up and down

**Comment by vidvidvid** [comment]
@alalonde I changed `skills_id` -> `skillIds` and `rolesId` -> `roleIds`, tested creating and editing quests in the client PR and everything works fine! :)

**Comment by vidvidvid** [comment]
> You can have multiple lines in a given migration, both up and down

@alalonde so should i join the quest role migrations?
`hasura/migrations/1641640073308_create_table_public_quest_role/up.sql`:
```
CREATE TABLE "public"."quest_role"("quest_id" UUID NOT NULL, "role" text NOT NULL, "rank" integer, PRIMARY KEY ("quest_id","role") , FOREIGN KEY ("quest_id") REFERENCES "public"."quest"("id") ON UPDATE restrict ON DELETE cascade, FOREIGN KEY ("role") REFERENCES "public"."PlayerRole"("role") ON UPDATE restrict ON DELETE restrict, UNIQUE ("quest_id"));
```

and `hasura/migrations/1642342313101_alter_table_public_quest_role_drop_constraint_quest_role_quest_id_key/up.sql`:

`alter table "public"."quest_role" drop constraint "quest_role_quest_id_key";`
by putting one below the other in the same file?

**Comment by alalonde** [comment]
> > You can have multiple lines in a given migration, both up and down
> 
> @alalonde so should i join the quest role migrations? `hasura/migrations/1641640073308_create_table_public_quest_role/up.sql`:
> 
> ```
> CREATE TABLE "public"."quest_role"("quest_id" UUID NOT NULL, "role" text NOT NULL, "rank" integer, PRIMARY KEY ("quest_id","role") , FOREIGN KEY ("quest_id") REFERENCES "public"."quest"("id") ON UPDATE restrict ON DELETE cascade, FOREIGN KEY ("role") REFERENCES "public"."PlayerRole"("role") ON UPDATE restrict ON DELETE restrict, UNIQUE ("quest_id"));
> ```
> 
> and `hasura/migrations/1642342313101_alter_table_public_quest_role_drop_constraint_quest_role_quest_id_key/up.sql`:
> 
> `alter table "public"."quest_role" drop constraint "quest_role_quest_id_key";` by putting one below the other in the same file?

Just an FYI, no need this time :)

**Review by alalonde — APPROVED** [review]

**Pull request merged**